voltdbclient/url.go

```diff
@@ -25,9 +26,23 @@ func parseURL(src string) (*url.URL, error) {
 			}
 		}
 	} else {
-		if u.Port() == "" {
+		if getPort(u.Host) == "" {
 			u.Host = u.Host + ":21212"
 		}
 	}
 	return u, nil
 }
+
+func getPort(host string) string{
```

Comment:
Thanks for the change, I was wondering why this snippet looks familiar then I remember I saw something  like this in the go standard library. Just wondering if we might need to give them proper credits.

Here is the snippet  https://github.com/golang/go/blob/master/src/net/url/url.go#L1025-L1037 

@nshi can you help us here, this looks like the optiomal choice. If we can't incorporate this without issues we can fallback to my suggested implementation.